src/operator/last.ts

```diff
@@ -17,11 +17,18 @@ import {EmptyError} from '../util/EmptyError';
  * @throws - Throws if no items that match the predicate are emitted by the source Observable.
  */
 export function last<T, R>(predicate?: (value: T, index: number, source: Observable<T>) => boolean,
-                           resultSelector?: (value: T, index: number) => R,
-                           defaultValue?: R): Observable<T> | Observable<R> {
+                           resultSelector?: (value: T, index: number) => R | void,
```

Comment:
Gotcha, since those to be dealt with projection signature in later, I assume it'd good to kick off this PR to be checked in first - what do you think? @masaeedu